### ISSUE TITLE
Goalie Tactic Assignment Failed in Tactic Assignment Test

### ISSUE THREAD

**Issue by jameschuang2002** [UBC-Thunderbots/Software#3180]
### Description of the task
While porting the deprecated stp_tests to reflect our current design when designing the auto-sub tactic, the tests with a goalie assigned failed unexpectedly. 

### Acceptance criteria
The tests in tactic_assignment_test.cpp passes successfully. 

- [ ] 

### Blocked By
#3031

**Comment by nycrat** [comment]
I actually have no idea what is going on with these tests:

<img width="1695" height="239" alt="Image" src="https://github.com/user-attachments/assets/cfc5d6a4-1417-436c-903d-ee0222ff0f4b" />

They aren't referred to in the code anywhere and have compile errors cause they use the old STP class. I'm suggesting we just get rid of these as a part of another issue. #3845
